tests/acceptance/ContactManagementCest.php

```diff
@@ -0,0 +1,207 @@
+        $I->waitForElementNotVisible('#MauticSharedModal-label', 30);
+
+        $I->amOnPage('/s/contacts');
+        $I->reloadPage(); // Ensure the latest data is loaded
+
+        // Search for the contact we just created
+        $I->fillField('#list-search', 'QuickAddFirstName');
+        $I->pressKey('#list-search', Facebook\WebDriver\WebDriverKeys::ENTER);
+        $I->waitForElementVisible('#leadTable', 10); // Wait for the search results to appear
+
+        // Verify the contact is in the list
+        $I->see('QuickAddFirstName', '#leadTable');
+
+        // Clear the search
+        $I->click('#btn-filter');
+    }
+
+    public function createContactFromForm(AcceptanceTester $I)
+    {
+        // Navigate to the contacts page
+        $I->amOnPage('/s/contacts');
+
+        // Click on "+New" button
+        $I->waitForElementClickable('#toolbar a:nth-child(2)', 30);
+        $I->click('#toolbar a:nth-child(2)');
+        $I->waitForText('New Contact', 30);
+
+        // Fill out the form fields
+        $I->waitForElementVisible('#lead_firstname', 10);
+        $I->fillField('#lead_firstname', 'FirstName');
+        $I->fillField('#lead_lastname', 'LastName');
+        $I->fillField('#lead_email', 'email@example.com');
+        $I->fillField('#lead_tags_chosen input', 'TestTag');
+        $I->pressKey('#lead_tags_chosen input', Facebook\WebDriver\WebDriverKeys::ENTER);
+
+        // Scroll back to the top of the page
+        $I->executeJS('window.scrollTo(0, 0);');
+
+        // Click the save and close button
+        $I->waitForElementClickable('#lead_buttons_save_toolbar', 30);
+        $I->click('#lead_buttons_save_toolbar');
+
+        // Wait for the contact details page to load
+        $I->waitForElementVisible('.page-header-title .span-block', 30);
+        $I->see('FirstName LastName', '.page-header-title .span-block');
+
+        // Click the close button on the contact details page
+        $I->waitForElementClickable('#toolbar > div.std-toolbar.btn-group > a:nth-child(3)', 30);
+        $I->click('a.btn.btn-default[href="/s/contacts"]');
+    }
+
+    public function acessEditContactFormFromList(AcceptanceTester $I)
+    {
+        // Navigate to the contacts page
+        $I->amOnPage('/s/contacts');
+
+        // Grab the name of the first contact in the list
+        $contactName = $I->grabTextFrom('#leadTable tbody tr:first-child td:nth-child(2) a div');
+
+        // Check we can see the first contact name
+        $I->see($contactName);
+
+        // Click on the dropdown caret on the first contact
+        $I->click('#leadTable tbody tr:first-child td:first-child div div button');
+
+        // Wait for the dropdown menu to show and click the delete menu option
+        $I->waitForElementClickable('#leadTable > tbody > tr:nth-child(1) > td:nth-child(1) > div > div > ul > li:nth-child(1) > a', 30);
```

Comment:
When I see selectors like this I wonder whether we shouldn't add unique IDs to those buttons instead. I know it will be extra work. But it could be helpful for the UI improvements that @andersonjeccel is working on as well. This is me thinking out loud. Not a request to implement it.

Reply:
Always add a unique ID to clickable things. This allow us to create keyboard shortcuts and other features later.

Reply:
This PR might help here https://github.com/mautic/mautic/pull/13911